var/ramble/repos/builtin/modifiers/apptainer/modifier.py

```diff
@@ -0,0 +1,289 @@
+# Copyright 2022-2024 The Ramble Authors
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+import re
+import os
+
+from ramble.modkit import *
+from ramble.util.hashing import hash_string
+
+import llnl.util.filesystem as fs
+
+
+class Apptainer(BasicModifier):
+    """Apptainer is a container platform. It allows you to create and run
+    containers that package up pieces of software in a way that is portable and
+    reproducible. You can build a container using Apptainer on your laptop, and
+    then run it on many of the largest HPC clusters in the world, local
+    university or company clusters, a single server, in the cloud, or on a
+    workstation down the hall. Your container is a single file, and you don’t
+    have to worry about how to install all the software you need on each
+    different operating system."""
+
+    container_extension = "sif"
+    name = "apptainer"
+
+    tags("container")
+
+    maintainers("douglasjacobsen")
+
+    mode("standard", description="Standard execution mode for apptainer")
+    default_mode("standard")
+
+    required_variable("container_name")
```

Comment:
nit: since this is likely the only place where these two variables are "defined", would be nice to have some descriptions on them.

Reply:
Added comments, and made a task to add descriptions to required variables, since that doesn't exist yet.